core/src/main/java/hudson/Util.java

```diff
@@ -1662,7 +1665,27 @@ public static int permissionsToMode(Set<PosixFilePermission> permissions) {
             throw new IOException(e);
         }
     }
-
+    
+    /**
+     * Compute the number of calendar days elapsed since the given date.
+     * As it's only the calendar days difference that matter, "11.00pm" to "2.00am the day after" returns 1,
+     * even if there are only 3 hours between. As well as "10am" to "2pm" both on the same day, returns 0.
+     */
+    @Restricted(NoExternalUse.class)
+    public static int differenceInCalendarDay(@Nonnull Date a, @Nonnull Date b){
+        LocalDate aLocal = a.toInstant().atZone(ZoneId.systemDefault()).toLocalDate();
+        LocalDate bLocal = b.toInstant().atZone(ZoneId.systemDefault()).toLocalDate();
+        return Period.between(aLocal, bLocal).getDays();
+    }
+    
+    /**
+     * @see #differenceInCalendarDay(Date, Date)
+     */
+    @Restricted(NoExternalUse.class)
+    public static int numOfCalendarDayToNow(@Nonnull Date date){
```

Comment:
Could name this something like `daysSince()` or `daysElapsedSince()`. Also, based on usage, you could also return 0 for future dates to simplify a bit.